docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,170 @@
+# Sonobuoy Tarball Client
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Objectives](#objectives)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+* [Unresolved Questions](#unresolved-questions)
+
+## Summary
+
+Sonobuoy delivers results in a gzipped tar archive. The structure of the tar
+is [documented here][tar structure] but programmatic access is up to every
+consumer to implement. The structure of the tarball can change between
+versions and consumer implementations become fragile.
+
+This proposal aims to describe an API for a client that can consumers can use
+to extract data from a Sonobuoy tarball.
+
+## Objectives
+
+Provide a friendly API for easy access to data found in Sonobuoy Tarballs
+across versions. Users should already be fairly familiar with the API below
+as it might look a lot like client-go. Instead of querying an API server we
+are querying a Sonobuoy tarball.
+
+### Goals
+
+ - Define and implement an API to access files in a Sonobuoy tarball.
+ - Support all tarball versions starting at v0.9.0.
+ - Have a familiar API.
+
+### Non-Goals
+
+- We will be unable to customize the tarball for every plugin as people can
```

Comment:
Could we make an interface plugin authors could plug in to? Maybe some high-level metadata we expect plugin authors to provide?

Reply:
Registering a callback for sub-parsing would be useful here.

Reply:
Maybe, yeah. We might tackle that at a later time, I think there is plenty of work to do and that's not really a use case we need to support right off the bat.